### PR TITLE
GOLINT: add comment for exported (checked by golint)

### PR DESCRIPTION
**Pull request by weiwei-xiao**
**What this PR does**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
add comment for exported (checked by golint)
```

### CODE REVIEW

File: common/proxy/proxy_factory.go [hunk 1]
```diff
@@ -22,10 +22,12 @@ import (
 	"github.com/apache/dubbo-go/protocol"
 )
 
+// ProxyFactory ...
 type ProxyFactory interface {
 	GetProxy(invoker protocol.Invoker, url *common.URL) *Proxy
 	GetAsyncProxy(invoker protocol.Invoker, callBack interface{}, url *common.URL) *Proxy
 	GetInvoker(url common.URL) protocol.Invoker
```
Comment (zouyx) [inline common/proxy/proxy_factory.go]: no comment for  interface's method?

File: common/url.go [hunk 2]
```diff
@@ -53,16 +53,19 @@ const (
 )
 
 var (
+	// DubboNodes ...
 	DubboNodes = [...]string{"consumers", "configurators", "routers", "providers"}
 	DubboRole  = [...]string{"consumer", "", "", "provider"}
```
Comment (zouyx) [inline common/url.go]: comment for DubboRole

File: common/url.go [hunk 3]
```diff
@@ -53,16 +53,19 @@ const (
...
 )
 
+// RoleType ...
 type RoleType int
 
 func (t RoleType) String() string {
```
Comment (zouyx) [inline common/url.go]: comment for String

File: config/application_config.go [hunk 4]
```diff
@@ -34,15 +35,22 @@ type ApplicationConfig struct {
 	Environment  string `yaml:"environment" json:"environment,omitempty" property:"environment"`
```
Comment (zouyx) [inline config/application_config.go]: comment for public variable?

File: config_center/zookeeper/listener.go [hunk 5]
```diff
@@ -27,32 +27,38 @@ import (
...
 	return &CacheListener{rootPath: rootPath}
 }
+
+// AddListener ...
 func (l *CacheListener) AddListener(key string, listener config_center.ConfigurationListener) {
 
 	// reference from https://stackoverflow.com/questions/34018908/golang-why-dont-we-have-a-set-datastructure
 	// make a map[your type]struct{} like set in java
-	listeners, loaded := l.keyListeners.LoadOrStore(key, map[config_center.ConfigurationListener]struct{}{listener: struct{}{}})
+	listeners, loaded := l.keyListeners.LoadOrStore(key, map[config_center.ConfigurationListener]struct{}{listener: {}})
```
Comment (zouyx) [inline config_center/zookeeper/listener.go]: why delete ```struct{}```?

File: cluster/cluster.go [hunk 6]
```diff
@@ -21,6 +21,7 @@ import (
 	"github.com/apache/dubbo-go/protocol"
 )
 
+// Cluster ...
 type Cluster interface {
 	Join(Directory) protocol.Invoker
```
Comment (zouyx) [inline cluster/cluster.go]: comment for interface method?

File: common/proxy/proxy_factory.go (hunk 1 above)
Comment (weiwei-xiao) [reply common/proxy/proxy_factory.go]: not hinted by golint

File: common/url.go (hunk 2 above)
Comment (weiwei-xiao) [reply common/url.go]: not hinted by golint. guess a bug of golint. add comment in next commit.

File: common/url.go (hunk 3 above)
Comment (weiwei-xiao) [reply common/url.go]: add comment in next commit

File: config/application_config.go (hunk 4 above)
Comment (weiwei-xiao) [reply config/application_config.go]: not hinted by golint.

File: config_center/zookeeper/listener.go (hunk 5 above)
Comment (weiwei-xiao) [reply config_center/zookeeper/listener.go]: `gofmt -s -w .` refactor the code.

File: cluster/cluster.go (hunk 6 above)
Comment (weiwei-xiao) [reply cluster/cluster.go]: not hinted by golint.

**Pull request merged**